tests/foreman/ui/test_contentview.py

```diff
@@ -3967,3 +3967,291 @@ def test_negative_non_readonly_user_actions(module_org, test_name):
         with raises(NavigationTriesExceeded) as context:
             session.contentview.search(cv.name)
         assert 'Navigation failed to reach [All]' in str(context.value)
+
+
+@tier2
+def test_positive_conservative_solve_dependencies(session, module_org):
+    """Performing solve dependencies on a package that is required by another
+    package.  Then performing solve dependencies on a root package with
+    its corresponding dependency.
+
+    :id: bffd50f3-77e4-492f-a0ac-09dfb95b1987
+
+    :setup:
+        1. Turn dependency solving to conservative.
+        2. Create and sync custom repo.
+        3. Create cv and add custom repo.
+        4. Create exclusion filter and exclude a package that is required
+        by another package (cockateel).
+        5. Publish CV
+        6. Check rpm package list for that version and make sure it's in the
+        list.
+        7. Create exclusion filter for a root package (duck) along with
+        package from above.
+        8. Republish and check rpm package list for new version and make
+        sure they are NOT in the list.
+
+    :expectedresults: In step 6, 'cockateel' should be in the list.
+        In step 8, 'duck' and 'cockateel' should NOT be in the list.
+
+    :CaseImportance: High
+    """
+    property_name = 'dependency_solving_algorithm'
+    param_value = 'conservative'
+    cv_name = gen_string('alpha')
+    repo_name = gen_string('alpha')
+    package1_name = 'duck'
+    package2_name = 'cockateel'
+    arch = 'noarch'
+    create_sync_custom_repo(
+        module_org.id,
+        repo_name=repo_name,
+        repo_url=FAKE_0_YUM_REPO
+    )
+    with session:
+        session.settings.update(
+            'name = {}'.format(property_name),
+            param_value
+        )
+        session.contentview.create({
+            'name': cv_name,
+            'solve_dependencies': True,
+        })
+        session.contentview.add_yum_repo(cv_name, repo_name)
+        filter_name = gen_string('alpha')
+        session.contentviewfilter.create(cv_name, {
+            'name': filter_name,
+            'content_type': FILTER_CONTENT_TYPE['package'],
+            'inclusion_type': FILTER_TYPE['exclude'],
+        })
+        session.contentviewfilter.add_package_rule(
+            cv_name, filter_name, package2_name, arch, ('All Versions'))
+        result = session.contentview.publish(cv_name)
+        assert result['Version'] == VERSION
+        packages = session.contentview.search_version_package(
```

Comment:
as you are searching singular package, why we are defining as 'packages'. can you please make it singular to 'package'. I got confused in between. This could apply the whole PR in all places.

Reply:
Sure I can change that.